aiida/cmdline/commands/cmd_import.py

```diff
@@ -25,6 +26,52 @@
 COMMENT_MODE = ['newest', 'overwrite']
 
 
+class ImportPath(click.Path):
+    """AiiDA Verdi implementation of Click's Path-type.
+    Introduce tests to distinguish whether the path is to a local file
+    or a URL string to a file that needs to be downloaded.
+    """
+
+    def __init__(self, **kwargs):
+        super(ImportPath, self).__init__(**kwargs)
+
+    def __call__(self, value, param=None, ctx=None):
+        if value is not None:
```

Comment:
is that check necessary here?
shouldn't we let covert handle that?

Reply:
I'm simply reimplementing the `__call__` function as it is for `click.Path`'s parent class `ParamType`. I think it's fine to handle it here and let `convert` handle the more detailed checks.